rust/core-lib/src/selection.rs

```diff
@@ -348,6 +350,17 @@ impl SelRegion {
     }
 }
 
+// Returns `[start..end)`
+impl<'a> RangeBounds<usize> for &'a SelRegion {
+    fn start_bound(&self) -> Bound<&usize> {
+        Bound::Included(&self.start)
```

Comment:
As I mentioned before, I'm not sure this is the right behavior, as start > end is allowed in a SelRegion (it means the region is reversed), but I'm expecting violates expectations of a RangeBounds. I could be wrong about this though.

Reply:
Good point; this should be resolved by making the start bound should be `self.start.min(self.end)`, and doing the inverse for the end bound?

Reply:
Yeah, implementing `start_bound` as `Bound::Included(min(&self.start, &self.end))` works just fine.